develop-docs/backend/api/design.mdx

```diff
@@ -1,62 +1,159 @@
+- **Do** use `snake_case` for path parameters. e.g. `tags/\{tag_name}/`.
+- **Do** consistently shorten parameters that are excessively long when the term will unambiguous. e.g. `organization` -> `org`.
 
-In the above example we flattened `projects`. The table below shows the existing flattened collections which works out with our existing APIs.
+Standard path parameters that should be shortened in routes:
 
-| First collection in URL | When to use | Parent | Identifier | Example |
-| --- | --------- | ----- | --- | --- |
-| organizations | When the resource cannot be attached to any other collection below parent like Project | N/A - always comes as first collection | \{organization_slug} | [Create a New Team](https://docs.sentry.io/api/teams/create-a-new-team/) |
-| teams | When the resource is under a specific team in the hierarchy | organizations | \{organization_slug}/ \{team_slug} | [Retreive Team Stats](https://docs.sentry.io/api/teams/retrieve-event-counts-for-a-team/) |
-| projects | When the resource is under a specific project in the hierarchy but not under an issue | organizations | \{organization_slug}/ \{project_slug} | [Create a New Client Key](https://docs.sentry.io/api/projects/create-a-new-client-key/)|
-| issues | When the resource is under a specific issue in the hierarchy | projects | \{issue_id} | [List an Issue's Events](https://docs.sentry.io/api/events/list-an-issues-events/)|
-| sentry-app-installations | When the resource is mapped to a specific integration | organizations | \{integration_slug} | [Delete an External Issue](https://docs.sentry.io/api/integration/delete-an-external-issue/)|
+- `organization` -> `org` (e.g. `/organizations/{org}/`)
 
-Here are some additional examples:
-- 👍 `/organizations/\{organization_slug}/projects/`
-- 👍 `/projects/\{organization_slug}/\{project_slug}/issues/`
-- 👎 `/projects/`
+### Prefixes and Scoping
+
+Information in Sentry is typically constrained by tenants. That is, almost all information is scoped to an organization. All endpoints which query customer data **must** be scoped to an organization:
+
+- **Do** prefix resource organizations collections with `organizations/\{org}`.
+- **Do** prefix resource project collections with `projects/\{org}/\{project}`.
+- **Do not** expose endpoints which require `org` as a query parameter (it should always be a path parameter).
+
+Knowing when to choose which constraint to couple an endpoint to will be based on the purpose of an endpoint. For example, if an endpoint is only ever going to be used \to query data for a single project, it should be prefixed with `/projects/{org}/{project}/things`. If an endpoint would need to exist to query multiple projects (which is common with cross-project queries), you likely should expose it as `/organizations/{org}/things`, and expose a query param to filter on the project(s).
```

Comment:
```suggestion
Knowing when to choose which constraint to couple an endpoint to will be based on the purpose of an endpoint. For example, if an endpoint is only ever going to be used to query data for a single project, it should be prefixed with `/projects/{org}/{project}/things`. If an endpoint would need to exist to query multiple projects (which is common with cross-project queries), you likely should expose it as `/organizations/{org}/things`, and expose a query param to filter on the project(s).
```

An extra slash, but otherwise this is great.